--- /dev/null
+++ ../src/ManagerRegistry.php
@@ -18,7 +18,7 @@
 
 	public function __construct(Connection $connection, EntityManagerInterface $em, Container $container)
 	{
-		$defaultConnection = $container->findByType(get_class($connection))[0];
+		$defaultConnection = $container->findByType(Connection::class)[0];
 		$defaultManager = $container->findByType(get_class($em))[0];
 
 		$connections = ['default' => $defaultConnection];
